### PR TITLE
Fix work of vttablet with non-empty vt_dba password.

### LINKED ISSUE #2154: ApplySchema and Backup fails if a dba password is set
After setting a password for a DBA user, we are seeing that both backups and ApplySchema command fails due to authentication issues:
1. with ApplySchema, it says that the password is used for authentication, but authentication fails
2. with backups, mysqladmin fails to authenticate with mysql and does not use the dba password provided to vttablet.

I had submitted the PR with respect to resolve this issue: https://github.com/youtube/vitess/pull/2148

However, it seems like it has not fixed the issue.

### PR DESCRIPTION
**Pull request by pivanof**
Apparently the --password flag on mysql and mysqladmin is in some way special
and it doesn't accept the value of password separeted from the flag by a space,
it requires it to be passed via '=' (--password=value). Another way to pass the
password is via short flag -p again with password following the flag without
space and without '='. I chose that method of passing password since it would
be consistent with flags -u and -S.

Also fixing the problem introduced in
https://github.com/youtube/vitess/pull/2148 when executeMysqlScript() would pass
vt_dba password even if it tries to connect as a different user.

Fixes #2154.

### CODE REVIEW

File: go/vt/mysqlctl/mysqld.go [hunk 1]
```diff
@@ -687,12 +687,12 @@ func deleteTopDir(dir string) (removalErr error) {
 
 // executeMysqlCommands executes some SQL commands,
 // using the mysql command line tool.
-func (mysqld *Mysqld) executeMysqlCommands(user, sql string) error {
-	return mysqld.executeMysqlScript(user, strings.NewReader(sql))
+func (mysqld *Mysqld) executeMysqlCommands(user, password, sql string) error {
```
Comment (alainjobart) [inline go/vt/mysqlctl/mysqld.go]: How about we pass in a *sqldb.ConnParams instead here?

File: go/vt/mysqlctl/schema.go [hunk 2]
```diff
@@ -230,7 +230,7 @@ func (mysqld *Mysqld) PreflightSchemaChange(dbName string, changes []string) ([]
 			initialCopySQL += s + ";\n"
 		}
 	}
-	if err = mysqld.executeMysqlCommands(mysqld.dba.Uname, initialCopySQL); err != nil {
+	if err = mysqld.executeMysqlCommands(mysqld.dba.Uname, mysqld.dba.Pass, initialCopySQL); err != nil {
```
Comment (alainjobart) [inline go/vt/mysqlctl/schema.go]: then this would just be (mysqld.dba, ...)

File: go/vt/mysqlctl/mysqld.go (hunk 1 above)
Comment (pivanof) [inline go/vt/mysqlctl/mysqld.go]: Done.

**Comment by alainjobart** [comment]
LGTM

**Pull request merged**